examples/complete/firestore/src/Todos.test.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+import { render, unmountComponentAtNode } from 'react-dom'
+import { act } from 'react-dom/test-utils'
+import { useSelector } from 'react-redux'
+import Todos from './Todos'
+
+jest.mock('react-redux')
+
+describe('Should test the sample component', () => {
+  let container = null
+  const expectedTodos = []
+
+  beforeEach(() => {
+    // setup a DOM element as a render target
+    container = document.createElement('div')
+    document.body.appendChild(container)
+  })
+
+  afterEach(() => {
+    // cleanup on exiting
+    unmountComponentAtNode(container)
+    container.remove()
+    container = null
+  })
+
+  it('show empty text when there is no text', () => {
+    useSelector.mockReturnValue(expectedTodos)
+    act(() => {
+      render(<Todos />, container)
+    })
+    const listItem = container.querySelectorAll('li')
+    expect(container.textContent).toBe('Todo list is empty')
+  })
+
+  it('should have on TodoItem when there is one to do', () => {
+    let moreTodo = {
+      id: 'todoId',
+      name: 'Sample todo',
+      text: 'This is for testing todos'
+    }
+    expectedTodos.push(moreTodo)
+    useSelector.mockReturnValue(expectedTodos)
+    act(() => {
+      render(<Todos />, container)
+    })
+    const listItems = container.querySelectorAll('li')
+        // this is not a real id, so only text which can be visible is Delete
```

Comment:
Not sure I understand this comment - what ID is it referring to? And wouldn't `Delete` be within the button in that list item?

Reply:
`TodoItem` in this example is using the `id` from the props to fetch the Todo from the firebase server. For the mock purpose, we have given a fake Id, which will not return any todo as it does not exist, only `delete` text will be shown. But we can imply that `TodoItem` is being rendered.

Reply:
Gotcha - thanks for the explanation